test/test-card-recorder.coffee

```diff
@@ -56,45 +63,93 @@ describe 'app.CardRecorder', ->
 
   describe '.getUpdateCards(callback)', ->
     sandbox = undefined
-    before ->
+    stub = undefined
+    error = null
+    cards = [ 'card1', 'card2', 'card3' ];
+
+    beforeEach ->
       sandbox = sinon.sandbox.create();
-      sandbox.stub(trello.prototype, 'get').returns(helpers.actionListMove)
+      stub = sandbox.stub(trello.prototype, 'get').yieldsAsync(error, cards)
       return
-    after ->
+
+    afterEach ->
       sandbox.restore()
+      error = new Error('Test Error')
+      return
+
+    it 'will ping trello and get the updated actions of a card', (done) ->
+      CR.getUpdateCards().then (resp) ->
+        expect(stub.callCount).to.eql 1
+        expect(resp).to.eql cards
+        done()
+        return
       return
-    it 'will ping trello and get the updated actions of a card', ->
-      CR.getUpdateCards "callback", (resp)->
-        expect(resp).to.equal(helpers.actionListMove)
+
+    it 'will survive a trello error', (done) ->
+      CR.getUpdateCards().catch (err) ->
+        expect(stub.callCount).to.eql 1
+        expect(err).to.eql error
+        done()
         return
       return
     return
 
   describe '.deleteCurrentComment(cardID)', ->
+    runs = 0
     sandbox = undefined
-    before ->
-      currentComment = _un.clone(helpers.mockCurrentComment)
-      currentComment["text"] = "**Current Stage** This comment says comment stage."
-      notCurrentComment = _un.clone(helpers.mockCurrentComment)
-      notCurrentComment["text"] = "This comment is not in the current stage."
+    getStub = undefined
+    delStub = undefined
+    error = new Error('Test Error')
+
+    beforeEach ->
+      currentComment = JSON.parse(JSON.stringify(helpers.mockCurrentComment))
+      currentComment.data.text = "**Current Stage** This comment says current stage."
```

Comment:
I realize it should be
`currentComment.data.text = "**Current Stage:** This comment says current stage."`